colorama/ansitowin32.py

```diff
@@ -80,12 +88,41 @@ def __init__(self, wrapped, convert=None, strip=None, autoreset=False):
         # create the proxy wrapping our output stream
         self.stream = StreamWrapper(wrapped, self)
 
+        conversion_supported = True
         on_windows = os.name == 'nt'
         # We test if the WinAPI works, because even if we are on Windows
         # we may be using a terminal that doesn't support the WinAPI
         # (e.g. Cygwin Terminal). In this case it's up to the terminal
         # to support the ANSI codes.
-        conversion_supported = on_windows and winapi_test()
+        if on_windows and platform.release().startswith('10'):
+            # Windows 10 supports ANSI sequences using virtual terminal mode.
+            # However, this has been turned off by default from an early Windows 10 version.
+            # We turn virtual terminal support back on, then set
+            # `conversion_supported` to False. Behaviour will then be the same
+            # as for Linux - colorama doesn't use win32 calls.
+            from ctypes import wintypes, byref  # import wintypes now we know we're on Windows.
+            old_mode = wintypes.DWORD()
+
+            stdout_handle = windll.kernel32.GetStdHandle(STDOUT)
+            # If the next call failed, we may be running on Windows pre 10.0.10586.
+            # Leave ANSI handling to colorama, there's no virtual terminal support available.
+            if windll.kernel32.GetConsoleMode(stdout_handle, byref(old_mode)):
+                if not (old_mode.value & ENABLE_VIRTUAL_TERMINAL_PROCESSING):
+                    if old_mode_value is None:
+                        # Don't set this again. __init__ gets called twice, for stdout and stderr.
+                        old_mode_value = old_mode.value
+                        print("Stored old_mode_value = " + str(old_mode_value))
+                    # Only enable if it's not already enabled
+                    print("enabling virtual terminal for " + str(self.stream))
+                    if windll.kernel32.SetConsoleMode(stdout_handle, old_mode_value | ENABLE_VIRTUAL_TERMINAL_PROCESSING):
```

Comment:
I am a bit reluctant to do this here. Normally "init" itself doesn't change any console properties, and everything is controllable with the init flags. This call is now made implicitly and in any case.

Reply:
I can't think of a cleaner way to do it, unfortunately. We're basically putting Windows into the same mode as Linux, and avoiding win32 calls. As far as I can see, that has to be done here.